JamGame/Assets/Scripts/TileUnion/Tile/TileImpl.cs

```diff
@@ -210,12 +207,9 @@ private CornerType ChooseCornerForCorridor(WallType[] walls)
             ) switch
             {
                 (true, _, _, true) => CornerType.Inside,
-                (true, _, true, _) => CornerType.WallRight,
-                (_, true, _, true) => CornerType.WallLeft,
-                (_, true, true, _) => CornerType.OutsideMiddle,
-                (true, true, _, _) => CornerType.OutsideRight,
-                (_, _, true, true) => CornerType.OutsideLeft,
-                _ => CornerType.None,
+                (true, _, _, _) => CornerType.WallRight,
+                (_, _, _, true) => CornerType.WallLeft,
+                (false, _, _, false) => CornerType.OutsideMiddle,
```

Comment:
It's less general than previous approach. Also remove duplication in this method and `ChooseCorner`

Reply:
Discussed